timelock-impl/src/main/java/com/palantir/atlasdb/timelock/batch/RemotingTimestampLeaseServiceAdapter.java

```diff
@@ -62,7 +63,7 @@ ListenableFuture<GetMinLeasedTimestampResponses> getMinLeasedTimestamps(
             Namespace namespace, GetMinLeasedTimestampRequests request, RequestContext context) {
         AsyncTimelockService service = getServiceForNamespace(namespace, context);
 
-        Map<TimestampLeaseName, ListenableFuture<Long>> futures = KeyedStream.of(request.get())
+        Map<TimestampLeaseName, ListenableFuture<Long>> futures = KeyedStream.of(Set.copyOf(request.get()))
```

Comment:
Feels annoying to do 3 copies here. Would you want to unwind this yourself into a for loop and ignore duplicates?

Reply:
Maybe streamex has a method that resolves duplicates?

Reply:
I think it has a `distinct`, just to avoid having both in the same package, I unrolled in a loop